packages/api/amplify_api/example/integration_test/rest_test.dart

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+import 'package:amplify_flutter/amplify_flutter.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:integration_test/integration_test.dart';
+
+import 'util.dart';
+
+const path = 'items';
+const expectedResponseText = 'Hello from Lambda!';
+
+void main({bool useExistingTestUser = false}) {
+  IntegrationTestWidgetsFlutterBinding.ensureInitialized();
+
+  HttpPayload _generateTestPayload() => HttpPayload.json({'name': 'mow lawn'});
+  void _validateRestResponse(AWSHttpResponse response) {
+    final body = response.decodeBody();
+    expect(response.statusCode, 200);
+    expect(body, expectedResponseText);
+  }
+
+  group('REST', () {
+    setUpAll(() async {
+      await configureAmplify();
+
+      if (!useExistingTestUser) {
+        await signUpTestUser();
+      }
+    });
+
+    tearDownAll(() async {
+      if (!useExistingTestUser) {
+        await deleteTestUser();
+      }
+    });
+
+    group('guest user access', () {
+      setUpAll(() async {
+        await signOutTestUser();
+      });
+
+      testWidgets('should send GET request', (WidgetTester tester) async {
```

Comment:
Is this unique to the backend? I can't see why a GET should succeed and a HEAD should fail for guest access. Shouldn't both fail?

Reply:
Good question, agree it's weird. It's part of this backend but I think more CLI thing that unique to this backend. In CLI it asks you if unauthenticated users can READ, WRITE, UPDATE, or DELETE. I said only yes for READ, authenticated users can do all, and this is the result. Not sure why they are mapping those CRUD to theseHTTP methods. Makes sense for everything except HEAD.